### PR TITLE
Update Checkbox.php

### PR DESCRIPTION
**Pull request by MoniqueHahnefeld**
## Description
This getFilterOptions seams little bit long-winded to me. In the parent is already inspected if ($idList || $usedOnly). So I would like to recommend this code :-) ... Am I wrong?

### CODE REVIEW

**Comment by stefanheimes** [comment]
You have to take care of the code style. Please take a look at Travis. I have copy the part from travis for you, this time.

``` 
phpcs:
    [phpcs] 
    [phpcs] FILE: ...ttribute_checkbox/src/MetaModels/Attribute/Checkbox/Checkbox.php
    [phpcs] ----------------------------------------------------------------------
    [phpcs] FOUND 9 ERRORS AFFECTING 7 LINES
    [phpcs] ----------------------------------------------------------------------
    [phpcs]   99 | ERROR | [x] Line indented incorrectly; expected 12 spaces,
    [phpcs]      |       |     found 11
    [phpcs]  100 | ERROR | [x] Line indented incorrectly; expected at least 16
    [phpcs]      |       |     spaces, found 14
    [phpcs]  101 | ERROR | [x] Line indented incorrectly; expected 12 spaces,
    [phpcs]      |       |     found 11
    [phpcs]  101 | ERROR | [x] Expected 1 space after closing brace; 0 found
    [phpcs]  101 | ERROR | [x] Expected 1 space after ELSE keyword; 0 found
    [phpcs]  102 | ERROR | [x] Line indented incorrectly; expected at least 16
    [phpcs]      |       |     spaces, found 12
    [phpcs]  103 | ERROR | [x] Line indented incorrectly; expected at least 16
    [phpcs]      |       |     spaces, found 12
    [phpcs]  104 | ERROR | [x] Line indented incorrectly; expected 12 spaces,
    [phpcs]      |       |     found 11
    [phpcs]  109 | ERROR | [x] Function closing brace must go on the next line
    [phpcs]      |       |     following the body; found 1 blank lines before
    [phpcs]      |       |     brace
```

File: src/MetaModels/Attribute/Checkbox/Checkbox.php [hunk 1]
```diff
@@ -93,35 +93,19 @@ public function getFieldDefinition($arrOverrides = array())
...
-                        $colName,
-                        $this->getMetaModel()->getTableName()
-                    )
-                );
-                while ($rows->next()) {
-                    $arrCount[$rows->$colName] = $rows->mm_count;
-                }
-            }
+        
+        $arrOptions = parent::getFilterOptions($idList, $usedOnly, $arrCount);
```
Comment (discordier) [inline src/MetaModels/Attribute/Checkbox/Checkbox.php]: 1. You are aware that `$arrOptions` is now `[]` when neither `$idList` nor `$usedOnly` is passed?
Before your changes it was `['', '1']`.
2. You killed the whole value counting when neither `$idList` nor `$usedOnly` is passed (which this whole block was for).

Please see the changes in d297bf85461df4e0434ccc4ebc32ad7fbaa82257 and really understand what they do and why they were added before removing all this code and half the functionality along the way.

File: src/MetaModels/Attribute/Checkbox/Checkbox.php [hunk 2]
```diff
@@ -93,35 +93,19 @@ public function getFieldDefinition($arrOverrides = array())
...
+              $arrOptions['1'] = $GLOBALS['TL_LANG']['MSC']['metamodelattribute_checkbox']['value_1'];
+           }else{
+            unset($arrOptions[$key]);
+            $arrOptions['0'] = $GLOBALS['TL_LANG']['MSC']['metamodelattribute_checkbox']['value_0'];
+           }
         }
+           
+        return $arrOptions;
 
-        // Finally use the correct language strings.
```
Comment (discordier) [inline src/MetaModels/Attribute/Checkbox/Checkbox.php]: Why did you remove this block and add your cumbersome loop above? Where was a problem?
Now you get problems if a key is i.e. `1`, you will add it as `'1'` again and have `['0', 1, '1']` as array keys in `$arrOptions` then. The original code does not have this flaw.
What did you try to achieve here?

Comment (MoniqueHahnefeld) [inline src/MetaModels/Attribute/Checkbox/Checkbox.php]: ok, I will check this and overthink this all.

**Comment by MoniqueHahnefeld** [comment]
@stefanheimes @discordier ok, I will check this, overthink this all and let you know...

File: src/MetaModels/Attribute/Checkbox/Checkbox.php (hunk 2 above)
Comment (MoniqueHahnefeld) [inline src/MetaModels/Attribute/Checkbox/Checkbox.php]: @discodier 
1. yes you are right. the '0' is a mistake. 
2. ok, what I don't really understand is why we have to count the amount of options. It's always a single checkbox with two options. And I suppose the Databasevalue of this attribute will only have the content : null, '', or '1'. So if a value is not a (string) 1 there went something wrong while the save action. Or how should the number 1 get into the database field?

Comment (discordier) [inline src/MetaModels/Attribute/Checkbox/Checkbox.php]: Although the second question rather belongs to the code block [above](https://github.com/MetaModels/attribute_checkbox/pull/18/files#r106331456) I'll answer here, as the question is here.

We are not counting the amount of options but the amount of ENTRIES per option.
Like: `['' => 450, 1 => 50]` which means that there are 450 items matching which are having a value of `''` and 50 items having a value of `1`.
